### ISSUE TITLE
Incorrect auto-correction for `Style/MinMaxComparison`

### ISSUE THREAD

**Issue by mishina2228** [rubocop/rubocop#11439]
Auto-correction for `Style/MinMaxComparison` breaks the code that contains `elsif`.

--------

## Expected behavior

I expect the following code:
```ruby
def foo(num)
  if num <= 0
    1
  elsif num > 100
    100
  else
    num
  end
end
```

to be corrected like this:
```ruby
def foo(num)
  if num <= 0
    1
  else
    [num, 100].min
  end
end
```

## Actual behavior

RuboCop produces this output:
```ruby
def foo(num)
  return unless num <= 0

  1
  [num, 100].min
end
```

<details><summary>detail</summary>

```
$ bundle exec rubocop sample.rb -A --debug
For /tmp/rubocop: configuration from /tmp/rubocop/.rubocop.yml
Default configuration from /home/owner/.rbenv/versions/3.2.0/lib/ruby/gems/3.2.0/gems/rubocop-1.43.0/config/default.yml
Use parallel by default.
Skipping parallel inspection: only a single file needs inspection
Inspecting 1 file
Scanning /tmp/rubocop/sample.rb
Loading cache from /home/owner/.cache/rubocop_cache/3c869baa22a9809daf22cb89594369b09ca7fbe0/6d7a3b621ca1730e04accd938619e4bdab66cfb1/1efd5c9777f48cd52021a16cbb2750a5a66bedc4
W

Offenses:

sample.rb:4:3: C: [Corrected] Layout/EmptyLineAfterGuardClause: Add empty line after guard clause.
  return unless num <= 0
  ^^^^^^^^^^^^^^^^^^^^^^
sample.rb:4:3: C: [Corrected] Style/GuardClause: Use a guard clause (return unless num <= 0) instead of wrapping the code inside a conditional expression.
  if num <= 0
  ^^
sample.rb:5:5: C: [Corrected] Layout/IndentationConsistency: Inconsistent indentation detected.
    1
    ^
sample.rb:6:3: C: [Corrected] Layout/IndentationConsistency: Inconsistent indentation detected.
  [num, 100].min
  ^^^^^^^^^^^^^^
sample.rb:6:3: W: Lint/Void: Literal 1 used in void context.
  1
  ^
sample.rb:6:3: C: [Corrected] Style/MinMaxComparison: Use [num, 100].min instead.
  elsif num > 100 ...
  ^^^^^^^^^^^^^^^
sample.rb:6:5: C: [Corrected] Layout/IndentationConsistency: Inconsistent indentation detected.
    [num, 100].min
    ^^^^^^^^^^^^^^
sample.rb:7:1: C: [Corrected] Layout/TrailingWhitespace: Trailing whitespace detected.
sample.rb:8:1: C: [Corrected] Layout/EmptyLinesAroundMethodBody: Extra empty line detected at method body end.

1 file inspected, 9 offenses detected, 8 offenses corrected
Finished in 0.12371505796909332 seconds
```

</details>

## Steps to reproduce the problem

Put the above code in a file and run `bundle exec rubocop sample.rb -A`.

## RuboCop version

```
$ bundle exec rubocop -V
1.43.0 (using Parser 3.2.0.0, rubocop-ast 1.24.1, running on ruby 3.2.0) [x86_64-linux]
```
